provider/pkg/provider/provider.go

```diff
@@ -1709,31 +1733,90 @@ func (k *azureNativeProvider) getAuthConfig() (*authentication.Config, error) {
 	return builder.Build()
 }
 
-func (k *azureNativeProvider) getAuthorizers(authConfig *authentication.Config) (tokenAuth autorest.Authorizer,
+func assertAzVersion() error {
+	const versionHint = "Please make sure the Azure CLI 2.37 or greater, but less than 3.x, is installed or configure another authentication method."
```

Comment:
Might be helpful to have a concrete resource to point them at on _how_ to configure another authentication method. This URL is one possible option, though a little long.
```suggestion
	const versionHint = "Please make sure the Azure CLI 2.37 or greater, but less than 3.x, is installed or configure another authentication method. See https://www.pulumi.com/registry/packages/azure-native/installation-configuration/#credentials for more information"
```